### PR TITLE
Add jetpack signature class to connection package

### PR DESCRIPTION
**Pull request by dereksmart**
#### Changes proposed in this Pull Request:
Adds `class.jetpack-signature.php` and class.jetpack-error.php to the Jetpack connection package.

**This changes a couple signature constants from define to class constant, not sure of the implications.**

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.

#### Testing instructions:

- Clean composer
- Rebuild 
- Look at Travis tests

#### Proposed changelog entry for your changes:

*N/A

### CODE REVIEW

File: _inc/lib/debugger/class-jetpack-cxn-tests.php [hunk 1]
```diff
@@ -284,13 +286,8 @@ protected function test__server_port_value() {
 		$http_ports  = array( 80 );
 		$https_ports = array( 80, 443 );
 
-		if ( defined( 'JETPACK_SIGNATURE__HTTP_PORT' ) ) {
```
Comment (mdawaffe) [inline _inc/lib/debugger/class-jetpack-cxn-tests.php]: These need to remain configurable constants. They are used for sites behind terminating proxies where the edge is running on a normal port, but the upstream is running on a weird port.

The correct fix is to use the `X-Forwarded-Port` header, but there are some sites that use these constants instead.

File: packages/connection/src/Manager.php [hunk 2]
```diff
@@ -266,6 +271,14 @@ public function disconnect_site() {
 
 	}
 
+	/**
+	 * @param $text
+	 * @return string
+	 */
+	function jetpack_sha1_base64( $text ) {
```
Comment (mdawaffe) [inline packages/connection/src/Manager.php]: Do we want this to have a `jetpack_` prefix?

File: packages/connection/legacy/class.jetpack-signature.php [hunk 3]
```diff
@@ -260,7 +258,3 @@ function join_with_equal_sign( $name, $value ) {
 		return "{$name}={$value}";
 	}
 }
-
-function jetpack_sha1_base64( $text ) {
```
Comment (mdawaffe) [inline packages/connection/legacy/class.jetpack-signature.php]: This removes a function from the global scope. Back compat?

**Review by mdawaffe — CHANGES_REQUESTED** [review]

File: packages/connection/src/Manager.php [hunk 4]
```diff
@@ -21,6 +21,11 @@ class Manager implements Manager_Interface {
 	const SECRETS_OPTION_NAME    = 'jetpack_secrets';
 	const MAGIC_NORMAL_TOKEN_KEY = ';normal;';
 	const JETPACK_MASTER_USER    = true;
+	// These constants can be set in wp-config.php to ensure sites behind proxies will still work.
```
Comment (mdawaffe) [inline packages/connection/src/Manager.php]: I don't think we need these at all anymore, do we?

Can we move the comments to jetpack.php and get rid of these class constants entirely?

Comment (roccotripaldi) [reply packages/connection/src/Manager.php]: how can we be sure that a plugin depending on this has set them?
Sure we are sure here in the newly minted mono-repo of jetpack
but outside of jetpack, can we know for certain?

File: packages/connection/legacy/class.jetpack-signature.php [hunk 5]
```diff
@@ -1,30 +1,27 @@
 <?php
 
-// These constants can be set in wp-config.php to ensure sites behind proxies will still work.
```
Comment (mdawaffe) [inline packages/connection/legacy/class.jetpack-signature.php]: Can we add these comments back to the new `define()`s in jetpack.php? I want it to remain clear that these constants are not a good way to solve the problem :)

**Comment by mdawaffe** [comment]
Nice work! Thanks for humoring my nitpicks.

**Pull request merged**